internal/witness/cmd/witness/impl/witness.go

```diff
@@ -104,7 +104,9 @@ func Main(ctx context.Context, opts ServerOpts) error {
 
 	glog.Infof("Starting witness server...")
 	srv := ih.NewServer(w)
-	r := mux.NewRouter()
+	// These options are required as some logID values contain forward
```

Comment:
As discussed above, it should be possible to generate URL-safe log IDs. We can discuss offline if this isn't clear (it could be that I'm wrong).

Reply:
See my comment above :).

Reply:
I'm not sure that I'm convinced. But you are, and you've spent a lot of time thinking about it. That's good enough for me :-)